### PR TITLE
NP-46264 Add category filter to dialogue

### PR DESCRIPTION
**Pull request by iBuzza**
# Description

Link to Jira issue: https://unit.atlassian.net/browse/NP-46264

In this PR:
- moved category selector/filter to own component
- added/changed said component in MyPage Dialogue and Advanced Search respectively
- added new prop to CategoryFilterDialogue.tsx to supply which search param to use
- Updates TicketSearchParams with new publicationType param

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes 
(note: On Advaned Search, the component might take a lot of horizontal space when multiple categories are selected, this is not a consequence of these changes though)

- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
 (note: unsure if the default chip should have a data-testid)

- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

### CODE REVIEW

File: src/components/CategorySearchFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router-dom';
+import { ResultParam, TicketSearchParam } from '../api/searchApi';
+import { CategoryFilterDialog } from '../pages/search/advanced_search/CategoryFilterDialog';
+import { PublicationInstanceType } from '../types/registration.types';
+import { CategoryChip } from './CategorySelector';
+
+interface CategorySearchFilterProps {
+  searchParam: ResultParam | TicketSearchParam;
```
Comment (iBuzza) [inline src/components/CategorySearchFilter.tsx:11]: Mulig dette skulle vært enda mer eksplisitt?

Comment (elkebab) [reply src/components/CategorySearchFilter.tsx:11]: Godt poeng! Det er vel i praksis bare disse atm?

```suggestion
  searchParam: ResultParam.CategoryShould | TicketSearchParam.PublicationType;
```

File: src/pages/my_page/MyPagePage.tsx [hunk 2]
```diff
@@ -86,6 +90,7 @@ const MyPagePage = () => {
     status: searchParams.get(TicketSearchParam.Status),
     viewedByNot: filterUnreadOnly && user ? user.nvaUsername : '',
     type: selectedTypesArray.join(','),
+    publicationType: publicationType.join(','),
```
Comment (elkebab) [inline src/pages/my_page/MyPagePage.tsx:93]: Ser ut som det gjøres samme som på PR i går her, hvor det gjøres en `.split()` bare for å gjøre en `.join()` etterpå. Er ikke dette det samme?
```suggestion
    publicationType: searchParams.get(TicketSearchParam.PublicationType),
```

File: src/pages/search/advanced_search/CategoryFilterDialog.tsx [hunk 3]
```diff
@@ -2,17 +2,23 @@ import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from '@mui/
...
-import { ResultParam } from '../../../api/searchApi';
+import { ResultParam, TicketSearchParam } from '../../../api/searchApi';
 import { CategorySelector } from '../../../components/CategorySelector';
 import { PublicationInstanceType } from '../../../types/registration.types';
 
 interface CategoryFilterDialogProps {
   open: boolean;
   currentCategories: PublicationInstanceType[];
   closeDialog: () => void;
+  searchParam: ResultParam | TicketSearchParam;
```
Comment (elkebab) [inline src/pages/search/advanced_search/CategoryFilterDialog.tsx:13]: Samme som du nevner med mer eksplisitt type her også da

**Review by elkebab — APPROVED** [review]

**Pull request merged**